docs/config_samples/dataset_types/landsat8_example_product.yaml

```diff
@@ -0,0 +1,48 @@
+name: landsat8_example_product
+description: Landsat 8 example product
+metadata_type: eo3
+
+metadata:
+    product:
+        name: landsat8_example_product
+    properties:
+        eo:instrument: OLI_TIRS
+        eo:platform: landsat-8
+
+measurements:
+    - name: 'red'
+      aliases: [band_4, sr_band4]
+      dtype: int16
+      nodata: -9999
+      units: 'reflectance'
+    - name: 'blue'
+      aliases: [band_2, sr_band2]
+      dtype: int16
+      nodata: -9999
+      units: 'reflectance'
+    - name: 'pixel_qa'
+      aliases: [pixel_quality, level2_qa]
+      dtype: uint16
+      nodata: 1
+      units: 'bit_index'
+      flags_definition:
+        pixel_qa:
+          bits: [0,1,2,3,4,5,6,7,8,9,10,11,12,13,14,15]
+          description: Level 2 pixel quality band
+          values:
+            1: Fill
+            2: Clear
+            4: Water
+            8: Cloud shadow
+            16: Snow
+            32: Cloud
+            64: Cloud Confidence Low Bit
+            128: Cloud Confidence High Bit
+            256: Cirrus Confidence Low Bit
+            512: Cirrus Confidence High Bit
+            1024: Terrain Occlusion
+            2048: Unused
```

Comment:
I don't think one can have duplicate names i.e. `Unused`. This whole flag definition is not "best practice", since values are a bit mask, even if only one bit is set at any given time.

Reply:
Aside from "Unused", can you point me to "best practice", please? For the most part this format is similar to that already described on this page in the 'flags_definition' section.

Reply:
Actually, it's only confusing since this is an enumerated type (one classification at a time) but values happen to be powers of 2 so it looks like a bit mask. For enumerated types above is a correct definition (apart for `Unused`) which you then use as `pixel_qa="Snow"`.

Since values are powers of two you could potentially define those too, such as

```yaml
      flags_definition:
        snow:
          bits: [4]
          description: ...
          values:
            16: True
            0: False
```
But there isn't much value in that since our masking only supports AND and not OR, and `pixel_qa="Snow"` vs `snow=True` is same same really. If we had something like `mask_one_of(..., snow=True, water=True)` then having those defined separately would be advantageous.

Let's just remove all `Unused` things and be done with that.

Reply:
Ok. Redundant `Unused`s removed. Took the liberty to:
- move the "match on other metadata" example from the rst into the yaml as a comment
- add a note for avoiding repeated names in flag_definition values
- add a shortened "bit on/off" example to the yaml
I figure examples and comments in the yaml are helpful as they are in the eo3 datasest example.